src/bank.rs

```diff
@@ -349,36 +359,61 @@ impl Bank {
         let mut ret = Vec::new();
         for (i, id) in ids.iter().enumerate() {
             if let Some(entry) = last_ids.sigs.get(id) {
-                if ((last_ids.nth - entry.2) as usize) <= MAX_ENTRY_IDS {
+                if ((last_ids.nth - entry.2) as usize) < MAX_ENTRY_IDS {
                     ret.push((i, entry.1));
                 }
             }
         }
         ret
     }
 
+    /// Looks through a list of tick heights and stakes, and finds the latest
+    /// tick that has achieved finality
+    pub fn get_finality_timestamp(
+        &self,
+        ticks_and_stakes: &mut [(u64, i64)],
+        supermajority_stake: i64,
+    ) -> Option<u64> {
+        // Sort by tick height
+        ticks_and_stakes.sort_by(|a, b| a.0.cmp(&b.0));
+        let last_ids = self.last_ids.read().unwrap();
+        let nth = last_ids.nth;
+        let mut total = 0;
+        for (tick_height, stake) in ticks_and_stakes.iter() {
+            if ((nth - tick_height) as usize) < MAX_ENTRY_IDS {
+                total += stake;
+                if total > supermajority_stake {
+                    return Self::tick_height_to_timestamp(&*last_ids, *tick_height);
```

Comment:
what is this "&*" convention?

Reply:
I forgot that the Deref trait takes care of this, simplified to just &, good catch!